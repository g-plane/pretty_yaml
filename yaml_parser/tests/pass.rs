use insta::{assert_snapshot, glob, Settings};
use std::{fs, path::Path};

#[test]
fn pass_snapshot() {
    glob!("pass/*.yaml", |path| {
        let input = fs::read_to_string(path).unwrap();
        let tree = match yaml_parser::parse(&input) {
            Ok(tree) => tree,
            Err(err) => panic!("failed to parse '{}': {err}", path.display()),
        };
        assert_eq!(
            tree.to_string(),
            input,
            "syntax tree of '{}' does not match source",
            path.display()
        );

        build_settings(path).bind(|| {
            let name = path.file_stem().unwrap().to_str().unwrap();
            assert_snapshot!(name, format!("{tree:#?}"));
        });
    });
}

fn build_settings(path: &Path) -> Settings {
    let mut settings = Settings::clone_current();
    settings.set_snapshot_path(path.parent().unwrap());
    settings.remove_snapshot_suffix();
    settings.set_prepend_module_to_snapshot(false);
    settings.remove_input_file();
    settings.set_omit_expression(true);
    settings.remove_input_file();
    settings.remove_info();
    settings
}
