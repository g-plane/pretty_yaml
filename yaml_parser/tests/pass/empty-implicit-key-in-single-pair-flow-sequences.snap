---
source: yaml_parser/tests/pass.rs
---
ROOT@0..42
  DOCUMENT@0..41
    BLOCK@0..41
      BLOCK_SEQ@0..41
        BLOCK_SEQ_ENTRY@0..17
          MINUS@0..1 "-"
          WHITESPACE@1..2 " "
          BLOCK@2..17
            FLOW@2..17
              FLOW_SEQ@2..17
                L_BRACKET@2..3 "["
                WHITESPACE@3..4 " "
                FLOW_SEQ_ENTRIES@4..16
                  FLOW_SEQ_ENTRY@4..15
                    FLOW_PAIR@4..15
                      COLON@4..5 ":"
                      WHITESPACE@5..6 " "
                      FLOW_MAP_VALUE@6..15
                        FLOW@6..15
                          PLAIN_SCALAR@6..15 "empty key"
                  WHITESPACE@15..16 " "
                R_BRACKET@16..17 "]"
        WHITESPACE@17..18 "\n"
        BLOCK_SEQ_ENTRY@18..41
          MINUS@18..19 "-"
          WHITESPACE@19..20 " "
          BLOCK@20..41
            FLOW@20..41
              FLOW_SEQ@20..41
                L_BRACKET@20..21 "["
                FLOW_SEQ_ENTRIES@21..40
                  FLOW_SEQ_ENTRY@21..40
                    FLOW_PAIR@21..40
                      COLON@21..22 ":"
                      WHITESPACE@22..23 " "
                      FLOW_MAP_VALUE@23..40
                        FLOW@23..40
                          PLAIN_SCALAR@23..40 "another empty key"
                R_BRACKET@40..41 "]"
  WHITESPACE@41..42 "\n"
